Sources/TuistKit/Generator/ProjectGenerator.swift

```diff
@@ -166,6 +171,34 @@ final class ProjectGenerator: ProjectGenerating {
         return nativeTargets
     }
 
+    fileprivate func generateTestTargetIdentity(project _: Project,
+                                                pbxproj: PBXProj,
+                                                pbxProject: PBXProject) {
+        func testTargetName(_ target: PBXTarget) -> String? {
+            guard let buildConfigurations = target.buildConfigurationList?.buildConfigurations else {
+                return nil
+            }
+
+            return buildConfigurations
+                .compactMap { $0.buildSettings["TEST_TARGET_NAME"] as? String }
+                .first
+        }
+
+        let testTargets = pbxproj.nativeTargets.filter { $0.productType == .uiTestBundle || $0.productType == .unitTestBundle }
+
+        for testTarget in testTargets {
+            guard let name = testTargetName(testTarget) else {
+                continue
+            }
+
+            guard let target = pbxproj.targets(named: name).first else {
+                continue
+            }
+
+            pbxProject.setTargetAttributes(["TestTargetID": target], target: testTarget)
```

Comment:
This will replace any existing attributes - I guess atm target attributes aren't being added elsewhere but will be hard to spot later on.

```swift
var attributes = pbxProject.targetAttributes[testTarget] ?? [:]
attributes["TestTargetID"] = target
pbxProject.setTargetAttributes(attributes, target: testTarget)
```

worth seeing if we can add a test for target attributes - will take a look